packages/addon-dev/src/rollup-incremental-plugin.ts

```diff
@@ -42,14 +53,22 @@ export default function incremental(): Plugin {
             continue;
           }
         }
-        if (bundle[checkKey]?.type === 'asset') {
-          if ((bundle[checkKey] as OutputAsset).source === generatedAssets.get(checkKey)) {
+        if (
+          bundle[checkKey]?.type === 'asset' &&
+          generatedAssets.has(checkKey)
+        ) {
+          if (
+            isEqual(
+              (bundle[checkKey] as OutputAsset).source,
+              generatedAssets.get(checkKey)
+            )
+          ) {
             delete bundle[key];
             continue;
           } else {
             generatedAssets.set(
               checkKey,
-              (bundle[checkKey] as OutputAsset).source
+              (bundle[checkKey] as OutputAsset).source.toString()
```

Comment:
Isn't that string casting breaking your `isEqual` implementation? Like when the plugin sees a jpeg asset, `(bundle[checkKey] as OutputAsset).source` will be a Buffer, but you put it into `generatedAssets` as a string, so `isEqual` will always return false, won't it?

Reply:
your right! thanks